test/crawlers/puppeteer_crawler.test.js

```diff
@@ -162,4 +164,104 @@ describe('PuppeteerCrawler', () => {
         expect(loadedCookies).toHaveLength(4);
         loadedCookies.forEach(cookie => expect(cookie).toEqual('TEST=12321312312'));
     });
+
+    describe('proxy', () => {
+        let requestList;
+        beforeEach(async () => {
+            requestList = new Apify.RequestList({
+                sources: [
+                    { url: 'http://example.com/?q=1' },
+                    { url: 'http://example.com/?q=2' },
+                    { url: 'http://example.com/?q=3' },
+                    { url: 'http://example.com/?q=4' },
+                ],
+            });
+            await requestList.initialize();
+        });
+
+        afterEach(() => {
+            requestList = null;
+        });
+
+        test('should work with proxyConfiguration', async () => {
```

Comment:
This only checks that passing `proxyConfiguration` does not break anything, but does not say anything about it working. Even the tests below only check if the `proxyInfo` is correct, but it does not check whether the browsers are actually started with the correct proxy.

Reply:
Added test for checking if the browser is launched with correctly passed `proxyUrl`.